web/server/entrypoint.go

```diff
@@ -249,6 +250,92 @@ func (s *robotServer) createWebOptions(cfg *config.Config) (weboptions.Options,
 	return options, nil
 }
 
+// A wrapper around actual config processing that also applies options from the
+// robot server.
+func (s *robotServer) processConfig(in *config.Config) (*config.Config, error) {
+	out, err := config.ProcessConfig(in)
+	if err != nil {
+		return nil, err
+	}
+	out.Debug = s.args.Debug || in.Debug
+	out.EnableWebProfile = s.args.WebProfile || in.EnableWebProfile
+	out.FromCommand = true
+	out.AllowInsecureCreds = s.args.AllowInsecureCreds
+	out.UntrustedEnv = s.args.UntrustedEnv
+	out.PackagePath = path.Join(viamDotDir, "packages")
+	return out, nil
+}
+
+// A function to be started as a goroutine that watches for changes, either
+// from disk or from cloud, to the robot's config. Starts comparisons based on
+// `currCfg`. Reconfigures the robot when config changes are received from the
+// watcher.
+func (s *robotServer) configWatcher(ctx context.Context, currCfg *config.Config, r robot.LocalRobot,
+	watcher config.Watcher,
+) {
+	// Reconfigure robot to have passed-in config before listening for any config
+	// changes.
+	r.Reconfigure(ctx, currCfg)
+
+	// Once reconfigure with initial config is complete; set initializing to
+	// false. Robot is now fully running and can indicate this through the
+	// MachineStatus endpoint.
+	r.SetInitializing(false)
```

Comment:
Because I see we had to expose a whole method for this on the `LocalRobot` API, I feel compelled to ask: would it be wrong to have `localRobot.Reconfigure` just set this value at the end?

Reply:
`newWithResources` calls `Reconfigure` to construct the robot, so if we set `initializing` to false at the end of `Reconfigure`, then `initializing` would be incorrectly false before the `Reconfigure` above completes.